### PR TITLE
Call start earlier in System.Net.Quic

### PR DESCRIPTION
**Pull request by jkotalik**
Still trying to run this test locally.

@scalablecory I didn't see an issue for this. Do you know if there is one?

### CODE REVIEW

**Comment by scalablecory** [comment]
I don't believe there is an issue for this yet.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs [hunk 1]
```diff
@@ -215,9 +205,10 @@ private async ValueTask<CancellationTokenRegistration> HandleWriteStartState(Can
             });
 
             // Make sure start has completed
-            if (shouldAwaitStart)
+            if (!_started)
             {
-                await _startTask.ConfigureAwait(false);
+                await _sendResettableCompletionSource.GetTypelessValueTask().ConfigureAwait(false);
```
Comment (halter73) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs]: I think we should guard better against attempted parallel writes. ManualResetValueTaskSourceCore will fail if you try to await it multiple times concurrently.

Also, why does this await have ConfigureAwait(false), but not other awaits?

Comment (jkotalik) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs]: I'll file an issue. Also I don't think I need ConfigureAwait(false)

Comment (jkotalik) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs]: https://github.com/dotnet/runtime/issues/32190

**Review by halter73 — APPROVED** [review]

**Pull request merged**